### DISCUSSION TITLE
Is there any system library needed for the overlay scrollbar?

### DISCUSSION THREAD

**Discussion by pastalian** [qutebrowser/qutebrowser#5986] [Q&A] [answered]
On my main machine (Gentoo Linux), overlay scrollbars appear as normal scrollbars.
I tried installing qutebrowser from
- package manager
- virtualenv (including dependencies)

and both does not work.

It works fine on my Fedora machine, so I'm wondering if my main machine doesn't have the required libraries or compile options.

## Version output
```
qutebrowser v1.14.1
Git commit: 
Backend: QtWebEngine (Chromium 83.0.4103.122)
Qt: 5.15.2

CPython: 3.9.1
PyQt: 5.15.2

sip: 4.19.24
colorama: 0.4.4
pypeg2: 2.15
jinja2: 2.11.2
pygments: 2.7.3
yaml: 5.3.1
cssutils: 1.0.2 $Id$
attr: 20.3.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.2
PyQt5.QtWebKitWidgets: no
pdf.js: 2.5.207 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.33.0
QtNetwork SSL: OpenSSL 1.1.1i  8 Dec 2020

Style: QFusionStyle
Platform plugin: xcb
OpenGL: NVIDIA Corporation, 4.6.0 NVIDIA 455.45.01
Platform: Linux-5.10.3-gentoo-x86_64-AMD_Ryzen_5_5600X_6-Core_Processor-with-glibc2.32, 64bit
Linux distribution: Gentoo/Linux (gentoo)
Frozen: False
Imported from /usr/lib/python3.9/site-packages/qutebrowser
Using Python from /usr/bin/python3.9
Qt library executable path: /usr/lib64/qt5/libexec, data path: /usr/share/qt5

Paths:
cache: /tmp/qutebrowser-basedir-djfegrbt/cache
config: /tmp/qutebrowser-basedir-djfegrbt/config
data: /tmp/qutebrowser-basedir-djfegrbt/data
runtime: /tmp/qutebrowser-basedir-djfegrbt/runtime
system data: /usr/share/qutebrowser

Autoconfig loaded: yes
Config.py: no config.py was loaded
Uptime: 0:00:03
```

**solved**: It seems the problem was casused by setting env variable `QTWEBENGINE_CHROMIUM_FLAGS`.
When this variable is set, some settings from qutebrowser to qtwebengine (overlay scrollbars, color scheme, ...)  are disabled.

**Comment by rien333** [comment, marked as answer]
I think you might have stumbled upon a chromium/qt bug here. I also can't get the overlay scrollbars to work, even though setting `c.scrolling.bar ` to any value other than `overlay` works fine (I have the same version of nearly all system libraries you have). It did work fine a few months ago.

Internally, qutebrowser just passes a bunch of flags to chromium when  the user indicates that they want to show overlay scrollbars. You could experiment with passing those flags directly (maybe check the web for what arguments chrome expects, this changes sometimes), by adding something like the following to your config.py: 

```
c.qt.args = […, 'enable-features=OverlayScrollbar,OverlayScrollbarFlashAfterAnyScrollUpdate,OverlayScrollbarFlashWhenMouseEnter"']
```

(Just typing this, I saw some overlay scrollbars pop up, will investigate further) 

**edit** Okay, weird,  ever since a recent update, overlay scrollbars appear on long codeblocks, eventhough I'm sure they not always did so.

**Reply by pastalian** [reply]
Solved! Thanks for your suggestion!
